featurebyte/feast/service/registry.py

```diff
@@ -287,7 +307,8 @@ async def update_document(
 
         if original_doc.registry_path:
             # remove old registry file
-            await self.storage.delete(Path(original_doc.registry_path))
+            with self.get_registry_storage_lock(timeout=FEAST_REGISTRY_REDIS_LOCK_TIMEOUT):
+                await self.storage.delete(Path(original_doc.registry_path))
 
         document = await self._move_registry_to_storage(recreated_model)
```

Comment:
If only this part is locked, could the registry document in mongo be out of sync with the registry in the storage? This might or might not matter, but I was under the impression that the lock would cover the whole duration of `update_document` to make the update operation atomic.